wolfcrypt/src/aes.c

```diff
@@ -2155,7 +2236,7 @@ int wc_AesSetIV(Aes* aes, const byte* iv)
 
         HAL_CRYP_Init(&hcryp);
 
-        while (sz > 0) {
+        while (len > 0) {
```

Comment:
Is > 0 the right comparison here?  If someone tries to decrypt 4 bytes we'll decrypt/write 16 bytes to output.